marlowe-dashboard-client/src/Page/Welcome/State.purs

```diff
@@ -93,74 +117,43 @@ that wallet: a `WalletCompanion` and a `MarloweApp`.
+-- case ajaxWalletDetails of
+--   Left ajaxError -> addToast $ ajaxErrorToast "Failed to generate wallet." ajaxError
+--   Right walletDetails -> do
+--     handleAction $ WalletNicknameInputAction $ InputField.Reset
+--     handleAction $ WalletNicknameInputAction $ InputField.SetValidator $ walletNicknameError walletLibrary
+--     assign _walletId $ walletDetails ^. _companionAppId
+--     handleAction $ OpenCard UseNewWalletCard
+{- [UC-WALLET-TESTNET-2][0] Restore a testnet wallet
 -}
-handleAction (WalletNicknameOrIdInputAction inputFieldAction) = do
-  toWalletNicknameOrIdInput $ InputField.handleAction inputFieldAction
-  case inputFieldAction of
-    InputField.SetValue walletNicknameOrId -> do
-      handleAction $ WalletNicknameOrIdInputAction $ InputField.SetValidator $ const Nothing
-      assign _remoteWalletDetails NotAsked
-      for_ (parsePlutusAppId walletNicknameOrId) \plutusAppId -> do
-        assign _remoteWalletDetails Loading
-        handleAction $ WalletNicknameOrIdInputAction $ InputField.SetValidator $ walletNicknameOrIdError Loading
-        ajaxWalletDetails <- lookupWalletDetails plutusAppId
-        assign _remoteWalletDetails $ fromEither ajaxWalletDetails
-        handleAction $ WalletNicknameOrIdInputAction $ InputField.SetValidator $ walletNicknameOrIdError $ fromEither ajaxWalletDetails
-        case ajaxWalletDetails of
-          Left _ -> pure unit -- feedback is shown in the view in this case
-          Right _ -> do
-            -- check whether this wallet ID is already in the walletLibrary ...
-            walletLibrary <- use _walletLibrary
-            assign _walletId plutusAppId
-            case findMin $ filter (\w -> w ^. _companionAppId == plutusAppId) walletLibrary of
-              Just { key } -> do
-                -- if so, open the UseWalletCard
-                handleAction $ WalletNicknameInputAction $ InputField.SetValue key
-                handleAction $ OpenCard UseWalletCard
-              Nothing -> do
-                -- otherwise open the UseNewWalletCard
-                handleAction $ WalletNicknameInputAction $ InputField.Reset
-                handleAction $ WalletNicknameInputAction $ InputField.SetValidator $ walletNicknameError walletLibrary
-                handleAction $ OpenCard UseNewWalletCard
-    InputField.SetValueFromDropdown walletNicknameOrId -> do
-      -- in this case we know it's a wallet nickname, and we want to open the use card
-      -- for the corresponding wallet
-      walletLibrary <- use _walletLibrary
-      for_ (lookup walletNicknameOrId walletLibrary) (handleAction <<< OpenUseWalletCardWithDetails)
-    _ -> pure unit
+handleAction RestoreTestnetWallet = do
+  walletName <- use (_walletNicknameInput <<< _value)
+  mnemonicPhraseStr <- use (_walletMnemonicInput <<< _value)
+  let
+    mnemonicPhrase = words mnemonicPhraseStr
+  result <- restoreWallet { walletName, mnemonicPhrase, passphrase: "" }
+  case result of
```

Comment:
Cosmetics: maybe some parts like `Milliseconds 1200.0` or even the whole `tell _submitButtonSlot "restore-wallet" <<< SubmitResult (Millisecond 1200.0)` could be extracted into a meaning full variable above?

Reply:
yeah, more than variable, a function with clearer meaning. Something like `informLoadingSubmitError` and `informLoadingSubmitSuccess`